### PR TITLE
RHAIENG-4347: chore(tests/browser): rename @stage tags to @tier, document quality gates

### PR DESCRIPTION
**Pull request by jiridanek**
https://redhat.atlassian.net/browse/RHAIENG-4347

## Summary
- Rename `@stage1`/`@stage2`/`@stage3` to `@tier1`/`@tier2`/`@tier3` in Playwright tests to align with the RHOAI TestOps quality gate standard agreed in `#wg-openshift-ai-quality` (Feb 2026).
- Add a "Test tags and quality gates" section to the browser tests README documenting what each tier means and when it runs.

## Context

The RHOAI TestOps quality gate standard defines:
- **Smoke**: very high / critical priority tests, every nightly build
- **Tier 1**: high-priority tests (excluding Smoke), daily on nightly builds
- **Tier 2**: medium/low-priority positive tests, weekly
- **Tier 3**: negative and destructive tests, weekly

The `@stage` naming was an interim choice; `@tier` is the org-wide standard used by all other component teams.

## Test plan
- [x] `pnpm exec playwright test --list --grep @tier1` lists the OCP console test
- [x] `pnpm exec playwright test --list --grep @smoke` still lists the OCP console test

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Documentation**
  * Added “Test tags and quality gates” guidance describing Playwright tag usage, tiered classifications (@smoke, @tier1, @tier2, @tier3) with cadence, and support for multiple orthogonal feature-area tags.

* **Tests**
  * Updated test metadata to adopt the new tier-based tag scheme for selecting/running tagged test suites.

### CODE REVIEW

File: tests/browser/README.md [hunk 1]
```diff
@@ -98,6 +98,28 @@ podman run --rm -t \
...
+
+| Tag | Gate | Meaning | Cadence |
+|---|---|---|---|
+| `@smoke` | Smoke | Very high / critical priority tests. Minimal validation that the component works at all. | Every nightly build |
+| `@tier1` | Tier 1 | High-priority tests (excluding Smoke). Core functionality and common user workflows. | Daily on nightly builds |
+| `@tier2` | Tier 2 | Medium/low-priority positive tests. Broader coverage, less critical paths. | Weekly |
+| `@tier3` | Tier 3 | Negative and destructive tests. Error handling, edge cases, recovery scenarios. | Weekly |
+
+A test that belongs to multiple tiers (e.g. a basic check that should run in every gate)
+can carry multiple tags: `{ tag: ['@smoke', '@tier1', '@tier2', '@tier3'] }`.
```
[the remark this answers is not in the document]
Comment (jiridanek) [reply tests/browser/README.md]: both applies, one is global guidance, and another is what we do in this repo

to resolve the contradiction, use judgement, case by case

**Review by atheo89 — APPROVED** [review]

File: tests/browser/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jiridanek) [reply tests/browser/README.md]: @coderabbitai recheck, fixed
